akka-remote/src/main/scala/akka/remote/artery/TaskRunner.scala

```diff
@@ -174,7 +182,9 @@ private[akka] class TaskRunner(system: ExtendedActorSystem, val idleCpuLevel: In
       case null         ⇒ // no command
       case Add(task)    ⇒ tasks.add(task)
       case Remove(task) ⇒ tasks.remove(task)
-      case Shutdown     ⇒ running = false
+      case Shutdown ⇒
+        running = false
+        shutdown.success(Done)
```

Comment:
`trySuccess` to be on the safe side, I think it can be called from the Aeron error handler also